components/eventing-controller/api/v1alpha1/subscription_conversion_test.go

```diff
@@ -0,0 +1,202 @@
+			alpha1Sub: NewDefaultSubscription(
+				WithFilter("app", OrderUpdatedEventType),
+				WithFilter("", OrderDeletedEventTypeNonClean),
+			),
+			alpha2Sub:  v1alpha2.NewDefaultSubscription(),
+			wantErrMsg: multipleSourceErrMsg,
+		},
+		{
+			name: "Converting NATS Subscription with Filters",
+			alpha1Sub: NewDefaultSubscription(
+				WithFilter(EventSource, OrderCreatedEventType),
+				WithFilter(EventSource, OrderUpdatedEventType),
+				WithFilter(EventSource, OrderDeletedEventTypeNonClean),
+				WithStatusCleanEventTypes([]string{
+					OrderCreatedEventType,
+					OrderUpdatedEventType,
+					OrderDeletedEventType,
+				}),
+			),
+			alpha2Sub: v1alpha2.NewDefaultSubscription(
+				v1alpha2.WithSource(EventSource),
+				v1alpha2.WithTypes([]string{
+					OrderCreatedEventType,
+					OrderUpdatedEventType,
+					OrderDeletedEventTypeNonClean,
+				}),
+				v1alpha2.WithStatusTypes([]v1alpha2.EventType{
+					{
+						OriginalType: OrderCreatedEventType,
+						CleanType:    OrderCreatedEventType,
+					},
+					{
+						OriginalType: OrderUpdatedEventType,
+						CleanType:    OrderUpdatedEventType,
+					},
+					{
+						OriginalType: OrderDeletedEventTypeNonClean,
+						CleanType:    OrderDeletedEventType,
+					},
+				}),
+				v1alpha2.WithStatusJetStreamTypes([]v1alpha2.JetStreamTypes{
+					{
+						OriginalType: OrderCreatedEventType,
+						ConsumerName: "",
+					},
+					{
+						OriginalType: OrderUpdatedEventType,
+						ConsumerName: "",
+					},
+					{
+						OriginalType: OrderDeletedEventTypeNonClean,
+						ConsumerName: "",
+					},
+				}),
+			),
+		},
+		{
+			name: "Converting BEB Subscription",
+			alpha1Sub: NewDefaultSubscription(
+				WithProtocolBEB(),
+				WithWebhookAuthForBEB(),
+				WithFilter(EventSource, OrderCreatedEventType),
+				WithFilter(EventSource, OrderUpdatedEventType),
+				WithFilter(EventSource, OrderDeletedEventTypeNonClean),
+				WithStatusCleanEventTypes([]string{
+					OrderCreatedEventType,
+					OrderUpdatedEventType,
+					OrderDeletedEventType,
+				}),
+				WithBEBStatusFields(),
+			),
+			alpha2Sub: v1alpha2.NewDefaultSubscription(
+				v1alpha2.WithSource(EventSource),
+				v1alpha2.WithTypes([]string{
+					OrderCreatedEventType,
+					OrderUpdatedEventType,
+					OrderDeletedEventTypeNonClean,
+				}),
+				v1alpha2.WithProtocolBEB(),
+				v1alpha2.WithWebhookAuthForBEB(),
+				v1alpha2.WithStatusTypes([]v1alpha2.EventType{
+					{
+						OriginalType: OrderCreatedEventType,
+						CleanType:    OrderCreatedEventType,
+					},
+					{
+						OriginalType: OrderUpdatedEventType,
+						CleanType:    OrderUpdatedEventType,
+					},
+					{
+						OriginalType: OrderDeletedEventTypeNonClean,
+						CleanType:    OrderDeletedEventType,
+					},
+				}),
+				v1alpha2.WithBEBStatusFields(),
+			),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
```

Comment:
nit: Since 2 methods are tested here I propose to use a **dedicated** `t.Run` with a **specific name** **for each method**.
This makes it easier in the case that the test fails to spot if there is a problem in converting from v1 -> v2 or the other way around :)

Reply:
Another idea would be to move the definition of the test cases outside of the method and have 2 different go test functions :)

Reply:
I am wondering why we don't have a controller integration test or something like that where the check that the conversion logic is correctly integrated ?

Reply:
I added t.Run for every conversion. I am not sure if we need the integration tests for that, because in that case we would test the kubebuilder logic and not our. Our is already covered by this unit test IMO.

Reply:
I don't agree. We would test the integration of both to ensure that our code is correctly integrated into the kubebuilder logic. Our conversion logic could be fully correct but still it not used in the webhook.